theseus/src/api/mod_sync.rs

```diff
@@ -0,0 +1,89 @@
+extern crate reqwest;
+
+use futures::StreamExt;
+
+use crate::mod_extraction::extract_info_from_jar;
+use crate::model::mod_type::JARLoadedMod;
+use crate::model::modrinth_mod_metadata::ModrinthModMetadata;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+/// Retrieve MR mod meta-data from https://docs.modrinth.com/api-spec/#tag/version-files/operation/versionFromHash.
+#[tracing::instrument]
+async fn retrieve_mod_metadata(file_hash: &String, hashing_alg: &String) -> crate::Result<ModrinthModMetadata> {
+    let url = format!("https://api.modrinth.com/v2/version_file/{}", 
+        file_hash);
+    let params = [("algorithm", hashing_alg)];
+
+    let url = reqwest::Url::parse_with_params(&url, &params)?;
+
+    Ok(crate::config::REQWEST_CLIENT
+        .get(url)
+        .send()
+        .await?
+        .error_for_status()?
+        .json::<ModrinthModMetadata>()
+        .await?)
+}
+
+
+fn list_jars(dir: &Path) -> Vec<String> {
+    let mut jars: Vec<String> = Vec::new();
+    if dir.is_dir() {
+        for entry in fs::read_dir(dir).unwrap() {
+            let entry = entry.unwrap();
```

Comment:
Don't unwrap here, use proper error handling

Reply:
This will fail if it fails to read an FS entry. You don't have to throw an error- an early return would also work